src/Maestro/Maestro.Web/Api/v2019_01_16/Controllers/GoalController.cs

```diff
@@ -0,0 +1,96 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Threading.Tasks;
+using Maestro.Data;
+using Maestro.Data.Models;
+using Maestro.Web.Api.v2019_01_16.Models;
+using Microsoft.AspNetCore.ApiVersioning;
+using Microsoft.AspNetCore.ApiVersioning.Swashbuckle;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.KeyVault;
+using Microsoft.EntityFrameworkCore;
+using System.Net;
+
+namespace Maestro.Web.Api.v2019_01_16.Controllers
+{
+    [Route("goals")]
+    [ApiVersion("2019-01-16")]
+    public class GoalController : ControllerBase
+    {
+        protected readonly BuildAssetRegistryContext _context;
+        public GoalController(BuildAssetRegistryContext context)
+        {
+            _context = context;
+        }
+
+        /// <summary>
+        ///   Exposes methods to set goal time <see cref="Goal"/>
+        /// </summary>
+        /// <param name="ChannelName"></param>
+        /// <param name="DefinitionId"></param>
+        /// <param name="GoalInMinutes"></param>
+        [HttpPost("{DefinitionId}/{ChannelName}")]
+        [SwaggerApiResponse(System.Net.HttpStatusCode.Created, Type = typeof(Models.Goal), Description = "Goal for given Channel and Definition-Id is created/updated")]
+        [ValidateModelState]
+        public virtual async Task<IActionResult> Create([Required] int DefinitionId, [Required] string ChannelName, [Required] int GoalInMinutes)
+        {
+            Data.Models.Channel channel = await _context.Channels
+                    .Where(c => c.Name.Equals(ChannelName)).FirstOrDefaultAsync();
```

Comment:
Do we maybe want this to be case insensitive? I'm not sure if that works in EF land. @alexperovich is my go-to expert for this. :-)

Reply:
It does work IIRC, but the database uses a case insensitive collation so it shouldn't matter.

Reply:
Of course, why should I expect string.Equals to follow the rules of string.Equals? I dislike EF so much...